feat: add includeIsolatedModules option

powlaa opened the pull request:
<!-- Thanks for taking the time to write this Pull Request ❤️ -->

## 🚀 Description
<!-- Describe your changes in detail -->
Adds `includeIsolatedModules` option (default: `false`) to the Module Graph plugin. When enabled, modules that have no dependencies or dependants will be shown as isolated Modules in the graph.

## 📄 Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
Fixes #56. Adds flexibility to configure the output graph to optionally show modules without connections to other modules.

## 🧪 How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->
- Added unit tests for checking the new configuration behaviour
- Tested with the sample project 

## 📦 Types of changes
<!-- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

powlaa committed:
feat: add include isolated modules option

Haratsu commented:
@iurysza is it possible that there is something wrong on the main branch?
I checked out the fork of @powlaa and ran tests on 
- mac, 
- ubuntu 24.04 
- as well as windows 11 

but there all gradle builds succeed.

I currently cannot find a way to reproduce the pre merge checks error.
https://github.com/iurysza/module-graph/actions/runs/17820457334/job/50771638665?pr=66

iurysza commented:
@Haratsu hmm, interesting. I haven't seen issues in the main branch.

I'll need to check this over the weekend when I'm back home.

Haratsu commented:
Thanks for your amazingly quick responses.
Also just locally build your main on each platform. without any issues in the tests. 
So I assume some fancy edge case within the github action setup or our everydays nightmare some encoding issue :wink:

Have a nice weekend

Haratsu commented:
Hello @iurysza,
Sorry for being a nuisance but did you have any further insight?

If you like we can have a RL-session in B. which is just ~2 hours from me. 
I could provide a coworking space as well. :grin:

iurysza commented:
Hey @Haratsu , no problem! I can reproduce the failure locally.

Running `./gradlew preMerge` fails on `DigraphCodeBuilderTest`. The issue seems to be that self-referential Gradle configuration dependencies (e.g., test classpaths) are being treated as isolated modules when `includeIsolatedModules=true`.

I'll try to push a fix for that. Sry for the delay.

Btw, are you in Berlin?

iurysza committed:
fix(DigraphBuilder): avoid self-referential dependencies in edge models

Haratsu commented:
@iurysza , Thanks for looking into it. 

I'm located in Dresden, but i have access to an office in Berlin ;-)

iurysza commented:
@Haratsu nice! It'd be nice hanging out is quite busy because of Droidcon Berlin. [Berlindroid](https://www.meetup.com/de-DE/berlindroid/events/310829745/?eventOrigin=group_events_list) is planning something for tomorrow.

Btw, pushed the fix. Now gradlew validation check is complaining. I'll need to check it tomorrow.

Haratsu commented:
@iurysza , I hope you had a great weekend.
Did you have a chance last week to check whats wrong with gradle validation?

Haratsu commented:
@iurysza 
I found the reason gradle wrapper validation action fails.
- the action checks whether the checksum of the gradle-wrapper.jar matches the expected value for the version references in gradle-wrapper.properties
- in commit bd23af7bdd40ef2ff3829da62db7eb6dd74747d3 you updated `gradle-wrapper.properties` but missed to update the gradle-wrapper.jar

powlaa committed:
fix: update gradle-wrapper.jar

powlaa commented:
Hey @iurysza, 
I have updated the gradle-wrapper.jar to match the `gradle-wrappers.properties` version. Can you check if this fixed the issue please?

iurysza committed:
fix: update gradle wrapper validation action to v4

The old gradle/wrapper-validation-action@v1 has been deprecated
in favor of gradle/actions/wrapper-validation@v4

iurysza commented:
Sry for the massive delay @powlaa and @Haratsu. Apparently we needed to update the GH action too.
It's out now in [v0.13.0](https://github.com/iurysza/module-graph/releases/tag/v0.13.0)
